addons/sale_timesheet/models/project_update.py

```diff
@@ -1,62 +1,68 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from collections import defaultdict
+
 from odoo import api, models
 from odoo.tools import float_utils, format_amount, formatLang
 
-from collections import defaultdict
-
 class ProjectUpdate(models.Model):
     _inherit = 'project.update'
 
     @api.model
     def _get_template_values(self, project):
+        template_values = super(ProjectUpdate, self)._get_template_values(project)
+        services = self._get_services_values(project)
+        profitability = self._get_profitability_values(project)
+        show_sold = template_values['project'].allow_billable and len(services.get('data', [])) > 0
+        show_profitability = profitability
         return {
-            **super(ProjectUpdate, self)._get_template_values(project),
-            'services': self._get_services_values(project),
-            'profitability': self._get_profitability_values(project),
+            **template_values,
+            'show_sold': show_sold,
+            'show_profitability': show_profitability,
+            'show_activities': template_values['show_activities'] or show_sold or show_profitability,
+            'services': services,
+            'profitability': profitability,
         }
 
     @api.model
     def _get_project_sols(self, project):
-        return self.env['sale.order.line'].search([('order_id', '=', project.sale_order_id.id), ('is_service', '=', True)])
+        # TODO: remove me in master
+	    return
 
     @api.model
     def _get_services_values(self, project):
         if not project.allow_billable:
             return {}
         services = []
-        total_sold_per_uom = defaultdict(float)
-        total_effective_per_uom = defaultdict(float)
         total_sold, total_effective, total_remaining = 0, 0, 0
-        sols = self._get_project_sols(project)
+        sols = project._get_sale_order_lines()
+        name_by_sol = dict(sols.name_get())
         for sol in sols:
             #We only want to consider hours and days for this calculation
-            if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id:
+            is_unit = sol.product_uom == self.env.ref('uom.product_uom_unit')
```

Comment:
I prefer store in a variable this expression: `self.env.ref('uom.product_uom_unit')`, **outside** the for loop to do once the `ref` method. 😇